### ISSUE TITLE
Sub Community filter is missing

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/micro-frontends-earn-app#5]
Sub Community filter is missing
<img width="1440" alt="Screenshot 2021-04-02 at 5 54 38 PM" src="https://user-images.githubusercontent.com/58783823/113415364-aaa59a80-93dc-11eb-862a-c0ca6eabc130.png">
<img width="1440" alt="Screenshot 2021-04-02 at 5 55 59 PM" src="https://user-images.githubusercontent.com/58783823/113415418-cc068680-93dc-11eb-8b27-404954b10601.png">

**Comment by sandhiyakavi** [comment]
@luizrrodrigues The sub community filter is still missing

![image](https://user-images.githubusercontent.com/55895538/114673375-15da5f80-9d41-11eb-9a28-12d7a653b1be.png)

**Comment by SathyaJayabal** [comment]
@luizrrodrigues @sushilshinde , filtering by multiple sub communities is not working (when an event and a group is selected).

Example url
https://platform.topcoder.com/earn/find/challenges?bucket=Open%20for%20Registration&`events[]=tco21&groups[]=b7f7c0f8-8ee8-409e-9e5c-33404983b635`&page=1&perPage=10&recommended=false&sortBy=updated&totalPrizesFrom=0&totalPrizesTo=10000&tracks[]=DES&tracks[]=DEV&tracks[]=DS&types[]=CH&types[]=F2F&types[]=TSK

Here there is an `AND` operation between groups and events which does not return any results
<img width="1440" alt="Screenshot 2021-04-21 at 10 11 58 AM" src="https://user-images.githubusercontent.com/58783823/115497823-06e83580-a28a-11eb-9b03-1cf5efaf01c7.png">

**Comment by Oanh-and-only-Oanh** [comment]
@SathyaJayabal, can you close this ticket?

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , filtering by multiple sub communities is not working
https://github.com/topcoder-platform/micro-frontends-earn-app/issues/5#issuecomment-823771293

**Comment by Oanh-and-only-Oanh** [comment]
ok thanks. escalating this to P0.

**Comment by luizrrodrigues** [comment]
@Oanh-and-only-Oanh @SathyaJayabal 

When selected a EVENT and a SUBCOMMUNITY, api will use `AND`, like:
if challenge is from group `TopGear` **AND** `TCO21`
or
if challenge is from event `TCO21` **AND** (from gropus `TopGear` **OR** `Veterans`)

So what we looking for here is?
if challenge is from group `TopGear` **OR** event `TCO21`

**Comment by Oanh-and-only-Oanh** [comment]
Correct, what we are looking for is **OR**. If challenge is from group Topgear OR event TCO21

**Comment by Oanh-and-only-Oanh** [comment]
@rootelement, can you help add this **`OR`** functionality to the api?

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , just a thought. What do you think about having two different filters, One for Events and another for Sub communities? It sort fo makes sense, as the TCOs are not sub communities.

**Comment by Oanh-and-only-Oanh** [comment]
Does that help by-pass the API issue?

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , it will. But we will need  to update design.

**Comment by Oanh-and-only-Oanh** [comment]
Ok, let me look into it.

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , that part is already working. So you can select multiple sub communities like zurich, topgear etc. Its only when you select a subcommunity and an event, we have the issue. We can also select multiple events, that too already works.

**Comment by luizrrodrigues** [comment]
>  just a thought. What do you think about having two different filters, One for Events and another for Sub communities? It sort fo makes sense, as the TCOs are not sub communities.

Agree

**Comment by sushilshinde** [comment]
@SathyaJayabal @luizrrodrigues should we close this one?

**Comment by SathyaJayabal** [comment]
@sushilshinde , we still have an issue with this filter. 
https://github.com/topcoder-platform/micro-frontends-earn-app/issues/5#issuecomment-825951755

We are thinking of 
https://github.com/topcoder-platform/micro-frontends-earn-app/issues/5#issuecomment-830205784

@Oanh-and-only-Oanh , can you confirm, if we can list events and sub communities as two different filters ?
